ArduPlane/mode_qloiter.cpp

```diff
@@ -132,25 +132,10 @@ void ModeQLoiter::run()
                                                                   quadplane.get_desired_yaw_rate_cds());
 
     if (plane.control_mode == &plane.mode_qland) {
-        if (poscontrol.get_state() < QuadPlane::QPOS_LAND_FINAL && quadplane.check_land_final()) {
-            poscontrol.set_state(QuadPlane::QPOS_LAND_FINAL);
-            quadplane.setup_target_position();
-#if AP_ICENGINE_ENABLED
-            // cut IC engine if enabled
-            if (quadplane.land_icengine_cut != 0) {
-                plane.g2.ice_control.engine_control(0, 0, 0, false);
-            }
-#endif  // AP_ICENGINE_ENABLED
-        }
         float height_above_ground = plane.relative_ground_altitude(plane.g.rangefinder_landing);
         float descent_rate_cms = quadplane.landing_descent_rate_cms(height_above_ground);
-
-        if (poscontrol.get_state() == QuadPlane::QPOS_LAND_FINAL && !quadplane.option_is_set(QuadPlane::OPTION::DISABLE_GROUND_EFFECT_COMP)) {
-            ahrs.set_touchdown_expected(true);
-        }
-
         pos_control->land_at_climb_rate_cm(-descent_rate_cms, descent_rate_cms>0);
-        quadplane.check_land_complete();
```

Comment:
`check_land_complete` returns if not in `QPOS_LAND_FINAL`, Qland never is.

https://github.com/ArduPilot/ardupilot/blob/863b6222de8a3a2f8b9ffe44ab64e148a337d64e/ArduPlane/quadplane.cpp#L3468-L3471